lib/ansible/modules/storage/netapp/na_ontap_snapshot.py

```diff
@@ -0,0 +1,320 @@
+        return
+
+    def create_snapshot(self):
+        """
+        Creates a new snapshot
+        """
+        snapshot_obj = netapp_utils.zapi.NaElement("snapshot-create")
+
+        # set up required variables to create a snapshot
+        snapshot_obj.add_new_child("snapshot", self.snapshot)
+        snapshot_obj.add_new_child("volume", self.volume)
+        # Set up optional variables to create a snapshot
+        if self.async_bool:
+            snapshot_obj.add_new_child("async", self.async_bool)
+        if self.comment:
+            snapshot_obj.add_new_child("comment", self.comment)
+        if self.snapmirror_label:
+            snapshot_obj.add_new_child(
+                "snapmirror-label", self.snapmirror_label)
+        try:
+            self.server.invoke_successfully(snapshot_obj, True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error creating snapshot %s: %s' %
+                                  (self.snapshot, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def delete_snapshot(self):
+        """
+        Deletes an existing snapshot
+        """
+        snapshot_obj = netapp_utils.zapi.NaElement("snapshot-delete")
+
+        # Set up required variables to delete a snapshot
+        snapshot_obj.add_new_child("snapshot", self.snapshot)
+        snapshot_obj.add_new_child("volume", self.volume)
+        # set up optional variables to delete a snapshot
+        if self.ignore_owners:
+            snapshot_obj.add_new_child("ignore-owners", self.ignore_owners)
+        if self.snapshot_instance_uuid:
+            snapshot_obj.add_new_child(
+                "snapshot-instance-uuid", self.snapshot_instance_uuid)
+        try:
+            self.server.invoke_successfully(snapshot_obj, True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error deleting snapshot %s: %s' %
+                                  (self.snapshot, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def modify_snapshot(self):
+        """
+        Modify an existing snapshot
+        :return:
+        """
+        snapshot_obj = netapp_utils.zapi.NaElement("snapshot-modify-iter")
+        # Create query object, this is the existing object
+        query = netapp_utils.zapi.NaElement("query")
+        snapshot_info_obj = netapp_utils.zapi.NaElement("snapshot-info")
+        snapshot_info_obj.add_new_child("name", self.snapshot)
+        query.add_child_elem(snapshot_info_obj)
+        snapshot_obj.add_child_elem(query)
+
+        # this is what we want to modify in the snapshot object
+        attributes = netapp_utils.zapi.NaElement("attributes")
+        snapshot_info_obj = netapp_utils.zapi.NaElement("snapshot-info")
+        snapshot_info_obj.add_new_child("name", self.snapshot)
+        snapshot_info_obj.add_new_child("comment", self.new_comment)
```

Comment:
supporting snapmirror-label would affect this logic, are only one of the two could be present, or both of them.